### ISSUE TITLE
Logging via TCP without source interface specified generates invalid EOS syntax. 

### ISSUE THREAD

**Issue by adietrich-ussignal** [aristanetworks/ansible-avd#1726]
### Issue Summary

After testing with the newest release (3.4.0), it does not appear that the fix generates a valid configuration that CVP can handle.

The only way I can get the ansible-avd playbook to complete successfully is by specifying this:
```
vrfs:
    management:
        source_interface: null
```

This appears to generate the following line of code in the configuration:

`logging source-interface None`

This subsequently causes CVP to mark this configuration line as an "error".

I believe the source-interface null should not generate a source-interface line in the configuration, which should eliminate this issue.

### Which component(s) of AVD impacted

eos_cli_config_gen

### How do you run AVD ?

Ansible CLI with AVD Runner

### Steps to reproduce

```shell
vrfs:
    management:
        source_interface: null
```
```


### Relevant log output

`logging source-interface None`


### Contributing Guide

- [X] I agree to follow this project's Code of Conduct

**Comment by UchihaItachiSama** [comment]
@adietrich-ussignal testing with the 3.4.0 using the following model 

```yaml
structured_config:
  logging:
    vrfs:
      MGMT:
        hosts: 
          30.30.30.7:
            protocol: tcp
          40.40.40.7:
            protocol: tcp
```

I can see the following configuration is generated
```yaml
logging vrf MGMT host 30.30.30.7 protocol tcp
logging vrf MGMT host 40.40.40.7 protocol tcp
```

which is accepted by the CLI. 

can you please share the model which you are using and the generated CLI command which is causing CVP to error out?

**Comment by adietrich-ussignal** [comment]
I removed the source-interface line from the configuration file, however, the workflow fails on the eos_cli_config_gen task.

```
TASK [arista.avd.eos_cli_config_gen : Generate eos intended configuration] *****
Friday 22 April 2022  12:05:43 +0000 (0:00:03.708)       0:02:30.933 ********** 
Friday 22 April 2022  12:05:43 +0000 (0:00:03.708)       0:02:30.933 ********** 
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: ansible.errors.AnsibleUndefinedVariable: 'dict object' has no attribute 'source_interface'
fatal: [SPINE-01]: FAILED! => changed=false 
  msg: 'AnsibleUndefinedVariable: ''dict object'' has no attribute ''source_interface'''
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: ansible.errors.AnsibleUndefinedVariable: 'dict object' has no attribute 'source_interface'
fatal: [SPINE-02]: FAILED! => changed=false 
  msg: 'AnsibleUndefinedVariable: ''dict object'' has no attribute ''source_interface'''
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: ansible.errors.AnsibleUndefinedVariable: 'dict object' has no attribute 'source_interface'
fatal: [DTS-01A]: FAILED! => changed=false 
  msg: 'AnsibleUndefinedVariable: ''dict object'' has no attribute ''source_interface'''
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: ansible.errors.AnsibleUndefinedVariable: 'dict object' has no attribute 'source_interface'
fatal: [DTS-01B]: FAILED! => changed=false 
...
```

**Comment by adietrich-ussignal** [comment]
Here is my model structure for reference:

```
logging:
  vrfs:
    management:
      hosts:
        host1.local:
          protocol: tcp
        host2.local:
          protocol: tcp
```

**Comment by ClausHolbechArista** [comment]
Can you share the output of the first task in the playbook? The one showing the version.
Similar to this:
```
TASK [arista.avd.eos_designs : Collection arista.avd version 3.4.0(git v3.4.0-27-g15b7451ed) loaded from /home/holbech/ansible-avd/ansible_collections] ***
fredag 22 april 2022  15:44:26 +0200 (0:00:00.486)       0:00:00.487 ********** 
```

**Comment by adietrich-ussignal** [comment]
That level of verbose output is not present in our playbook run. Here's a bit more insight as to how we pull down the repo. 

```
/usr/bin/git -c protocol.version=2 fetch --no-tags --prune --progress --no-recurse-submodules --depth=1 origin +refs/heads/releases/v3.x.x*:refs/remotes/origin/releases/v3.x.x* +refs/tags/releases/v3.x.x*:refs/tags/releases/v3.x.x*

/usr/bin/git checkout --progress --force -B releases/v3.x.x refs/remotes/origin/releases/v3.x.x

TASK [arista.avd.eos_designs : Create required output directories if not present] ***
```

**Comment by ClausHolbechArista** [comment]
That explains it then. You are still on v3.3.x. From v3.4.0 we are no longer using the releases/v3.x.x branch. I suggest that you install from Ansible Galaxy instead of GitHub. This way you will always get the latest stable release. `ansible-galaxy collection install arista.avd`

**Comment by adietrich-ussignal** [comment]
Using the new AVD-All-in-One container, this is no longer an issue and has been resolved in 3.4.0.
